src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -3132,6 +3141,7 @@ public void onSelection(SelectionEvent<Integer> event)
       if (event.getSelectedItem() >= 0)
       {
          activeEditor_ = editors_.get(event.getSelectedItem());
+         consoleEditorHadFocusLast_ = false;
```

Comment:
I'm wondering if we think it's okay to let these changes wait until
v0.9.1000 -- are the current APIs problematic for a significant class of
plugins?


On Fri, Jan 22, 2016 at 9:09 PM, Kevin Ushey <notifications@github.com>
wrote:

> In
> src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java
> <https://github.com/rstudio/rstudio/pull/558#discussion_r50613605>:
>
> > @@ -3132,6 +3141,7 @@ public void onSelection(SelectionEvent<Integer> event)
> >        if (event.getSelectedItem() >= 0)
> >        {
> >           activeEditor_ = editors_.get(event.getSelectedItem());
> > +         consoleEditorHadFocusLast_ = false;
>
> This was the 'best' spot I could find for hooking into changes in the
> active editor. It's a bit asymmetric, but noting that:
>
>    1. We listen for console focus events -- when the console editor is
>    focused, consoleEditorHadFocusLast_ is set to true;
>    2. When a source editor is selected, we run through this bit of code
>    (setting the activeEditor_ in the process), and also set
>    consoleEditorHadFocusLast_ to false.
>
> It feels a bit unnatural but it seems to work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/558/files#r50613605>.
>